src/server/messages.ts

```diff
@@ -1,17 +1,19 @@
-import * as vscode from 'vscode';
 import { ServerResult } from '../client/server_result';
 import { SOAPClient } from '../client/SOAP_client';
 
+// The interface of functions which have to send information to the server
+// (for convenience in typing in other files)
+export type ServerFunction = (filename: string, project: string, text: string) => Promise<ServerResult>;
+
 export async function validate(filename: string, project: string, text: string): Promise<ServerResult> {
     const client = new SOAPClient(filename, project, text);
-    try {
-        const result = await client.executeCommand("atf");
-        return result;
-    } catch (err) {
-        const errMsg = `An error has occurred:
-            ${err}
-            The log may contain more details.
-            `;
-        vscode.window.showErrorMessage(errMsg);
-    }
+    const result = await client.executeCommand("atf");
+    return result;
```

Comment:
Why is the error handling gone?

Reply:
It's moved to `workWithServer` in `extension.ts` (the calling function). The old error handling code returned `undefined` after showing the warning message, which caused an error later in `handleResult`. I thought about having an "empty" result value, but it seemed more complicated than letting the error float up.